### PR TITLE
Add H2O

### PR DESCRIPTION
**Pull request by ledell**
[https://github.com/h2oai/awesome-h2o](https://github.com/h2oai/awesome-h2o)

This is a curated list of all the awesome projects, applications, research, tutorials, courses and books that use the open source H2O distributed machine learning platform.  

- We curate this list, so it's not simply a list of all the H2O content we find -- it's just the awesome stuff.  
- It is also not just a list of our own tutorials & content (that's listed somewhere else), this is a list of entirely community created works that we find on the web.  
- The "Awesome H2O" list has been around almost a year, so by now there's a decent amount of content on there and we feel it's ready to be included in the Awesome list.
- The list items do not follow the suggested `- [Fish](…) - User-friendly shell.` format because the content items (blog post titles, research paper titles, book titles) usually have long, self-descriptive names.  Instead hyperlinked the title and used a internally self-consistent format to put dates after blog post titles, years after books titles, and authors associated with research papers, looking [here](https://github.com/terryum/awesome-deep-learning-papers) for inspiration.

# By submitting this pull request I confirm I've read and complied with the below requirements.

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

- I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title.<br>For example, `Add Name of List`, not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with these requirements:
	- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
	- It's the result of hard work and the best I could possibly produce.
	- Non-generated Markdown file in a GitHub repo.
	- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
	- Not a duplicate.
	- Only has awesome items. Awesome lists are curations of the best, not everything.
	- Includes a project logo/illustration whenever possible.
		- Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
		- The image should link to the project website or any relevant website.
		- The image should be high-DPI. Set it to maximum half the width of the original image.
	- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
	- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
		- Should be placed on the right side of the readme heading.
		- Should link back to this list.
	- Has a Table of Contents section.
		- Should be named `Contents`, not `Table of Contents`.
		- Should be the first section in the list.
		- Should only have one level of sub-lists, preferably none.
	- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
		- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
		- If you use a license badge, it should be SVG, not PNG.
	- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
		- The file should be named `contributing.md`. Casing is up to you.
	- Has consistent formatting and proper spelling/grammar.
		- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
		- The description starts with an uppercase character and ends with a period.
		- Drop all the `A` / `An` prefixes in the descriptions.
		- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
	- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Go to the top and read it again.

### CODE REVIEW

**Comment by ledell** [comment]
@legodude17 I've added a description of H2O in the first paragraph.  Thanks!

**Comment by legodude17** [comment]
I think descriptions for the papers and books would be a good idea.

**Comment by ledell** [comment]
@legodude17 Similar to the "Blog posts/Tutorials" section, the books and papers already have long names that are self-descriptive, so repeating that a second time seems redundant to me.  If you wanted a longer description than what the title already provides, I could copy/paste the paper abstracts (~paragraph), but that seems way too long.  We include author names and publication dates in addition to title, so we already have more information than most other awesome lists that include books and papers/tutorials.  

It seems like the `- [Fish](…) - User-friendly shell.` format makes a lot of sense for list items that are not self descriptive, e.g. `Node.js - JavaScript runtime built on Chrome's V8 JavaScript engine.` but just adds redundancy to items that are self-descriptive book and papers titles.  Is it okay to keep what we have?

**Comment by legodude17** [comment]
The main bit I was thinking about is how it was unclear what some of the papers have to do with h2o.

**Comment by ledell** [comment]
@legodude17 All the papers listed here use H2O to produce results for their research or application.  Example: "Deep learning and association rule mining for predicting drug response in cancer " uses H2O Deep Learning to predict drug response in cancer.

The entry added to the awesome list README helps clarify this: 

[H2O](https://github.com/h2oai/awesome-h2o) - Research, tutorials and applications that use the H2O distributed machine learning platform.

File: readme.md [hunk 1]
```diff
@@ -268,6 +268,7 @@
 	- [Tutorials](https://github.com/ujjwalkarn/Machine-Learning-Tutorials#readme)
 	- [ML with Ruby](https://github.com/arbox/machine-learning-with-ruby#readme) - Learning, implementing, and applying Machine Learning using Ruby.
 	- [Core ML Models](https://github.com/likedan/Awesome-CoreML-Models#readme) - Models for Apple's machine learning framework.
+	- [H2O](https://github.com/h2oai/awesome-h2o) - Research, tutorials and applications that use the H2O distributed machine learning platform.
```
Comment (sindresorhus) [inline readme.md]: > The entry in the Awesome list should:
Include a short description about the project/theme of the list. It should not describe the list itself.

The description should be about H2O itself, not the list.

**Comment by sindresorhus** [comment]
> If you use a license badge, it should be SVG, not PNG.

⬆️

**Comment by sindresorhus** [comment]
Only a couple of minor things left and this is ready to be merged :)

**Comment by sindresorhus** [comment]
Ping :)

**Comment by ledell** [comment]
@sindresorhus Thanks for the ping. I have made the requested edits, so hopefully it's ready to go!: [1](https://github.com/sindresorhus/awesome/pull/1125/commits/b8cd80e59c90ddb62eb1814aacd596a24d5a537f) [2](https://github.com/h2oai/awesome-h2o/commit/c0ee4b3627429e60fb7eb0b0cc627d790bf77eb5)

**Comment by sindresorhus** [comment]
The descriptions in the Courses section needs to end in a dot.

**Comment by sindresorhus** [comment]
Merging this now as it has been open for a very long time. Thanks for your patience. Don't forget to fix the above point.

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1046140014814326784

**Comment by ledell** [comment]
@sindresorhus I fixed the last point.  Thank you for merging & tweeting!

**Pull request merged**